lib/util.js

```diff
@@ -403,8 +403,10 @@ function formatValue(ctx, value, recurseTimes) {
   // Some type of object without properties can be shortcutted.
   if (keys.length === 0) {
     if (typeof value === 'function') {
-      return ctx.stylize(`[Function${value.name ? `: ${value.name}` : ''}]`,
-                         'special');
+      const asyncPrefix = binding.isAsyncFunction(value) ? 'async ' : '';
+      return ctx.stylize(
+        `[${asyncPrefix}Function${value.name ? `: ${value.name}` : ''}]`,
```

Comment:
Maybe. I didn't think about that. We don't do it for GeneratorFunction though. Should I extend the PR to support it too?